src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -1381,6 +1388,13 @@ else if (ppolicy == ProjectionPolicy.NONE && nativeCRS != null) {
                 schema = typeBuilder.buildFeatureType();
             }
 
+            // applying wrappers using
+            // implementations of org.geoserver.catalog.RetypeFeatureTypeCallback
+            for (RetypeFeatureTypeCallback callback :
+                    GeoServerExtensions.extensions(RetypeFeatureTypeCallback.class)) {
+                fs = (SimpleFeatureSource) callback.wrapFeatureSource(info, fs);
```

Comment:
You should not assume this will always by a ``SimpleFeatureSource`, check this method returning type.

Reply:
will add a check